data_prep/generate_annual_averages.py

```diff
@@ -0,0 +1,191 @@
+import os, os.path
+import re
+import logging
+import sys
+
+from argparse import ArgumentParser
+from datetime import datetime
+
+from cdo import Cdo
+from netCDF4 import Dataset, num2date, date2num
+import numpy as np
+from dateutil.relativedelta import relativedelta
+
+from Cmip5File import Cmip5File
+
+log = logging.getLogger(__name__)
+logging.basicConfig(stream=sys.stdout, level=4)
+
+def iter_matching(dirpath, regexp):
+    # http://stackoverflow.com/questions/4639506/os-walk-with-regex
+    """Generator yielding all files under `dirpath` whose absolute path
+       matches the regular expression `regexp`.
+       Usage:
+
+           >>> for filename in iter_matching('/', r'/home.*\.bak'):
+           ....    # do something
+    """
+    for dir_, dirnames, filenames in os.walk(dirpath):
+        for filename in filenames:
+            abspath = os.path.join(dir_, filename)
+            if regexp.match(abspath):
+                yield abspath
+
+def create_annual_avg_file(in_fp, out_fp, variable):
+    '''
+    Generates file with annual averages for the given variable across
+    all time within an input file
+
+    Parameters:
+        in_fp: input file path
+        out_fp: output file path
+        variable (str): name of the variable which is being processed
+
+    '''
+    supported_vars = {
+        'cddETCCDI', 'csdiETCCDI', 'cwdETCCDI', 'dtrETCCDI', 'fdETCCDI',
+        'gslETCCDI', 'idETCCDI', 'prcptotETCCDI', 'r10mmETCCDI', 'r1mmETCCDI',
+        'r20mmETCCDI', 'r95pETCCDI', 'r99pETCCDI', 'rx1dayETCCDI',
+        'rx5dayETCCDI', 'sdiiETCCDI', 'suETCCDI', 'thresholds', 'tn10pETCCDI',
+        'tn90pETCCDI', 'tnnETCCDI', 'tnxETCCDI', 'trETCCDI', 'tx10pETCCDI',
+        'tx90pETCCDI', 'txnETCCDI', 'txxETCCDI', 'wsdiETCCDI', 'tasmin',
+        'tasmax', 'pr'
+    }
+
+    if variable not in supported_vars:
+        raise Exception("Unsupported variable: can't yet process {}".format(variable))
+
+    if not os.path.exists(os.path.dirname(out_fp)):
+        os.makedirs(os.path.dirname(out_fp))
+
+    cdo = Cdo()
+    cdo.yearmean(input=in_fp, output=out_fp)
+
+def update_annual_avg_file_global_metadata(nc):
```

Comment:
One of my big pet peeves is naming a function or a task "update [something or other]". It really doesn't provide any description as to what the function is supposed to do or what the expected result is. One may as well name the function `randomly_change_global_metadata()` :) because all we know from the name is that something will be different afterwards.